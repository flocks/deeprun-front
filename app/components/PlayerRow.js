import React, { PropTypes } from 'react';
import Radium from 'radium';
import global from '../styles/global';
import UICard from './UICard';

const PlayerRow = ({ ...props }) => {
    return (
        <div style={style.base} onClick={props.onClick}>
            <div>
                <UICard card={props.cards[0]} />
                <UICard card={props.cards[1]} />
            </div>

            <p style={global.label}>{props.label}</p>
            {(props.equity) ?
                <div style={style.equity}> {props.equity}%</div>
            :
                <div style={[style.equity, style.equity_null]}>%</div>
            }
        </div>
    );
};

PlayerRow.defaultProps = {
    cards: [{}, {}],
    equity: null,
    onClick: function() {}
};

PlayerRow.propTypes = {
    label: PropTypes.string.isRequired,
    cards: PropTypes.arrayOf(PropTypes.object),
    equity: PropTypes.number,
    onClick: PropTypes.func
};

const style = {
    base: {
        textAlign: 'left',
        position: 'relative',
        marginBottom: '20px',

    },
    equity: {
        color: 'grey',
        position: 'absolute',
        fontSize: '35px',
        fontFamily: 'Montserrat',
        fontWeight: '800',
        right: '25px',
        top: '5px'
    },
    equity_null: {
        opacity: '0.1'
    }
};

export default Radium(PlayerRow);
