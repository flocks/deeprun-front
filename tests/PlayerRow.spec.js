import React from 'react';
import renderer from 'react-test-renderer';
import PlayerRow from '../app/components/PlayerRow';


describe('<PlayerRow />', () => {
    it('should render properly with empty value', () => {
        const component = renderer.create(
            <PlayerRow
                label="player 1"
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render properly with cards', () => {
        const cards = [{rank: '8', suit: 'diamond'}, {rank: '7', suit: 'diamond'}];
        const component = renderer.create(
            <PlayerRow
                label="player 1"
                equity={45}
                cards={cards}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
