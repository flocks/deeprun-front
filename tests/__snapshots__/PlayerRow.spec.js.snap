// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerRow /> should render properly with cards 1`] = `
<div
  data-radium={true}
  onClick={[Function]}
  style={
    Object {
      "marginBottom": "20px",
      "position": "relative",
      "textAlign": "left",
    }
  }
>
  <div
    data-radium={true}
  >
    <div
      data-radium={true}
      onClick={[Function]}
      style={
        Object {
          "background": "#5B97DF",
          "borderColor": "#5B97DF",
          "borderStyle": "solid",
          "borderWidth": "4px",
          "boxSizing": "border-box",
          "color": "white",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Montserrat",
          "fontSize": "25px",
          "fontWeight": "800",
          "height": "54px",
          "lineHeight": "45px",
          "marginRight": "5px",
          "textAlign": "center",
          "transition": "all 200ms linear",
          "verticalAlign": "bottom",
          "width": "40px",
        }
      }
    >
      <div
        style={undefined}
      >
        8
      </div>
    </div>
    <div
      data-radium={true}
      onClick={[Function]}
      style={
        Object {
          "background": "#5B97DF",
          "borderColor": "#5B97DF",
          "borderStyle": "solid",
          "borderWidth": "4px",
          "boxSizing": "border-box",
          "color": "white",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Montserrat",
          "fontSize": "25px",
          "fontWeight": "800",
          "height": "54px",
          "lineHeight": "45px",
          "marginRight": "5px",
          "textAlign": "center",
          "transition": "all 200ms linear",
          "verticalAlign": "bottom",
          "width": "40px",
        }
      }
    >
      <div
        style={undefined}
      >
        7
      </div>
    </div>
  </div>
  <p
    data-radium={true}
    style={
      Object {
        "color": "#A09D9D",
        "fontSize": "13px",
        "fontStyle": "italic",
        "marginTop": "5px",
      }
    }
  >
    player 1
  </p>
  <div
    data-radium={true}
    style={
      Object {
        "color": "grey",
        "fontFamily": "Montserrat",
        "fontSize": "35px",
        "fontWeight": "800",
        "position": "absolute",
        "right": "25px",
        "top": "5px",
      }
    }
  >
     
    45
    %
  </div>
</div>
`;

exports[`<PlayerRow /> should render properly with empty value 1`] = `
<div
  data-radium={true}
  onClick={[Function]}
  style={
    Object {
      "marginBottom": "20px",
      "position": "relative",
      "textAlign": "left",
    }
  }
>
  <div
    data-radium={true}
  >
    <div
      data-radium={true}
      onClick={[Function]}
      style={
        Object {
          "background": "#D7D8D7",
          "borderColor": "#D7D8D7",
          "borderStyle": "solid",
          "borderWidth": "4px",
          "boxSizing": "border-box",
          "color": "white",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Montserrat",
          "fontSize": "25px",
          "fontWeight": "800",
          "height": "54px",
          "lineHeight": "45px",
          "marginRight": "5px",
          "textAlign": "center",
          "transition": "all 200ms linear",
          "verticalAlign": "bottom",
          "width": "40px",
        }
      }
    >
      <div
        style={undefined}
      />
    </div>
    <div
      data-radium={true}
      onClick={[Function]}
      style={
        Object {
          "background": "#D7D8D7",
          "borderColor": "#D7D8D7",
          "borderStyle": "solid",
          "borderWidth": "4px",
          "boxSizing": "border-box",
          "color": "white",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Montserrat",
          "fontSize": "25px",
          "fontWeight": "800",
          "height": "54px",
          "lineHeight": "45px",
          "marginRight": "5px",
          "textAlign": "center",
          "transition": "all 200ms linear",
          "verticalAlign": "bottom",
          "width": "40px",
        }
      }
    >
      <div
        style={undefined}
      />
    </div>
  </div>
  <p
    data-radium={true}
    style={
      Object {
        "color": "#A09D9D",
        "fontSize": "13px",
        "fontStyle": "italic",
        "marginTop": "5px",
      }
    }
  >
    player 1
  </p>
  <div
    data-radium={true}
    style={
      Object {
        "color": "grey",
        "fontFamily": "Montserrat",
        "fontSize": "35px",
        "fontWeight": "800",
        "opacity": "0.1",
        "position": "absolute",
        "right": "25px",
        "top": "5px",
      }
    }
  >
    %
  </div>
</div>
`;
